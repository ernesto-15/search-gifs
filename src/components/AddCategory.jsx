import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './styles/AddCategory.css';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((prevCategories) => [inputValue, ...prevCategories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        className="input-category"
        type="text"
        value={inputValue}
        onChange={handleChange}
        placeholder="Search for any gif"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
