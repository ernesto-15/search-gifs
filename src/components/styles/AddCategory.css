.input-category {
  width: 100%;
  padding: 20px 10px;
  font-size: 1.3rem;
  color: gray;
  background-color: #09203f;
  border: none;
  border-radius: 20px;
}

form {
  margin: 35px auto;
}

@media (min-width: 768px) {
  form {
    max-width: 1000px;
  }
}
