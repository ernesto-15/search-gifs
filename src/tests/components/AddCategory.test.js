import React from 'react';
import AddCategory from '../../components/AddCategory.jsx';
import { shallow } from 'enzyme';

describe('tests in AddCategory component', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change input', () => {
    const input = wrapper.find('input');
    const value = 'hello world';
    input.simulate('change', { target: { value } });
  });

  test('should not post categories on submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories function and clean input', () => {
    const value = 'hello world';

    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
