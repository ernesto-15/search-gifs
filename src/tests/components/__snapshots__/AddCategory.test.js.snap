// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in AddCategory component should show component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input-category"
    onChange={[Function]}
    placeholder="Search for any gif"
    type="text"
    value=""
  />
</form>
`;
